### PR TITLE
ref(ts): convert DateSummary 

### PR DESCRIPTION
**Pull request by davidenwang**
converts to typescript

### CODE REVIEW

File: src/sentry/static/sentry/app/components/organizations/timeRangeSelector/dateSummary.tsx [hunk 1]
```diff
@@ -10,39 +9,26 @@ import {DEFAULT_DAY_END_TIME, DEFAULT_DAY_START_TIME} from 'app/utils/dates';
...
-    /**
-     * Start date value for absolute date selector
-     * Accepts a JS Date or a moment object
-     *
-     * React does not support `instanceOf` with null values
-     */
-    start: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
+type Props = {
+  start: Date;
+  end: Date;
```
Comment (scttcper) [inline src/sentry/static/sentry/app/components/organizations/timeRangeSelector/dateSummary.tsx:14]: This most likely should be `type RelaxedDateType = string | number | Date;` like timeSince. The proptypes also allowed strings

Comment (markstory) [reply src/sentry/static/sentry/app/components/organizations/timeRangeSelector/dateSummary.tsx:14]: We have a type `DateString` that could work here, it doesn't include `number` currently. It is used in a bunch of places that interact with moment, and query string values. moment also has `moment.MomentInput` which could work here.

File: src/sentry/static/sentry/app/components/organizations/timeRangeSelector/dateSummary.tsx [hunk 2]
```diff
@@ -10,39 +9,26 @@ import {DEFAULT_DAY_END_TIME, DEFAULT_DAY_START_TIME} from 'app/utils/dates';
...
   }
 
-  formatTime(date, withSeconds = false) {
+  formatTime(date: Date, withSeconds = false) {
     return this.getFormattedDate(date, `HH:mm${withSeconds ? ':ss' : ''}`);
   }
 
   render() {
-    const {className, start, end} = this.props;
+    const {start, end} = this.props;
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/organizations/timeRangeSelector/dateSummary.tsx:31]: wasn't `className` being used?

File: src/sentry/static/sentry/app/components/organizations/timeRangeSelector/dateSummary.tsx (hunk 1 above)
Comment (davidenwang) [reply src/sentry/static/sentry/app/components/organizations/timeRangeSelector/dateSummary.tsx:14]: ooh gotcha yea I'll opt for DateString then

File: src/sentry/static/sentry/app/components/organizations/timeRangeSelector/dateSummary.tsx (hunk 2 above)
Comment (davidenwang) [reply src/sentry/static/sentry/app/components/organizations/timeRangeSelector/dateSummary.tsx:31]: Hmmm from what I can see DateSummary is only used in one place and the only props it receives are start and end, but I could be missing something.

File: src/sentry/static/sentry/app/components/organizations/timeRangeSelector/dateSummary.tsx (hunk 1 above)
Comment (davidenwang) [reply src/sentry/static/sentry/app/components/organizations/timeRangeSelector/dateSummary.tsx]: ooh gotcha yea I'll just use moment.MomentInput then

**Review by markstory — APPROVED** [review]

**Pull request merged**